netweaver/monitoring.sls

```diff
@@ -1,26 +1,36 @@
 {%- from "netweaver/map.jinja" import netweaver with context -%}
 {% set host = grains['host'] %}
 
+
 prometheus_sap_host_exporter_pkg:
   pkg.installed:
     - name: prometheus-sap_host_exporter
 
+# the sid, instance number pair of a node is unique, so we need to adapt configuration
+{% for node in netweaver.nodes if host == node.host %}
+
+{% set instance = node.instance %}
+{% set instance = '{:0>2}'.format(node.instance) %}
+{% set daemon_instance = '{}_{}'.format(node.sid, instance) %}
+
+# don't install exporter for db instance
+{%- if not node.sap_instance == "db" %}
+
 sap_host_exporter_exporter_service:
   service.running:
-    - name: prometheus-sap_host_exporter
+    - name: prometheus-sap_host_exporter@{{ daemon_instance }}
     - enable: True
     - restart: True
     - require:
       - pkg: prometheus_sap_host_exporter_pkg
-      - file: sap_host_exporter_configuration
+      - file: sap_host_exporter_configuration_{{ daemon_instance }}
```

Comment:
Besides the `require` entry , you need to add the `file` state in a `watch` state in salt. Have a look in:
https://docs.saltstack.com/en/latest/ref/states/all/salt.states.service.html

Reply:
done thx